encrypted-media/resources/retrieve-persistent-license.html

```diff
@@ -0,0 +1,92 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <meta name="timeout" content="long">
+    <title>Encrypted Media Extensions: Retrieve stored persistent-usage-record</title>
+    <link rel="help" href="https://w3c.github.io/encrypted-media/">
+
+    <!-- Helper scripts for Encrypted Media Extensions tests  -->
+    <script src=/encrypted-media/util/utils.js></script>
+    <script src=/encrypted-media/util/fetch.js></script>
+    <script src=/encrypted-media/util/utf8.js></script>
+    <script src=/encrypted-media/util/testmediasource.js></script>
+
+    <!-- Message handler for Clear Key keysystem -->
+    <script src=/encrypted-media/util/clearkey-messagehandler.js></script>
+
+  </head>
+  <body>
+    <div id='log'></div>
+
+    <div id='video'>
+      <video id="videoelement" width="200px"></video>
+    </div>
+
+    <script>
+
+    window.addEventListener( 'message', function( event ) {
+
+        var config = event.data.config,
+            configuration = {   initDataTypes: [ config.initDataType ],
+                                audioCapabilities: [ { contentType: config.audioType } ],
+                                videoCapabilities: [ { contentType: config.videoType } ],
+                                sessionTypes: [ 'persistent-license' ] },
+            assertions = [ ];
+
+            var _mediaKeySession;
+
+            config.video = document.getElementById('videoelement');
+            config.messagehandler = messagehandler;
+
+        function onFailure(error) {
+            assertions.push( { actual: false, expected: true, message: error.toString() } );
+            window.opener.postMessage(assertions, '*');
+        }
+
+        function onMessage( event )
+        {
+            assertions.push( { expected: true, actual: event instanceof window.MediaKeyMessageEvent, message: "event is of correct class" } );
+            assertions.push( { expected: 'message', actual: event.type, message: "event type is message" } );
+
+            assert_in_array( event.messageType,
+                            [ 'license-request', 'individualization-request', 'license-release' ] );
+
+            config.messagehandler( config.keysystem, event.messageType, event.message )
+            .then( function( response ) {
+                event.target.update( response ).catch(onFailure);
+            });
+        }
+
+        function onTimeupdate(event) {
+            if ( config.video.currentTime > ( config.duration || 5 ) ) {
+                config.video.pause();
+                window.opener.postMessage(assertions, '*');
```

Comment:
It seems we should be more explicit about the test being completed successfully. Maybe there should be an explicit value to avoid any false successes.